src/nupic/types/BasicType.cpp

```diff
@@ -581,10 +581,10 @@ void BasicType::convertArray(void *ptr1, NTA_BasicType toType, const void *ptr2,
     default:
       break;
     }
-  } catch (nupic::Exception e) {
+  } catch (nupic::Exception &e) {
```

Comment:
I believe this is for g++-8, not c++17?

Reply:
Its possible, I made both changes (to c++17 and to gcc-8) at the same time